cluster.py

```diff
@@ -1,21 +1,46 @@
 import logging
+import socket
+import time
 from pathlib import Path
-from typing import Dict
+from typing import Dict, Tuple
 
 from ccmlib import scylla_cluster as ccm
 
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
 
+def get_cluster_id() -> Tuple[socket.socket, int]:
+    """gets integer number that will be used for ip prefix of cluster nodes,
+    so it's possible to run tests in parallel.
+
+    Returns tuple of socket and cluster id, where socket later needs to be closed."""
+    logger.info("Getting cluster id for support parallel tests...")
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    for cluster_id in range(126):
+        try:
+            sock.bind((f'127.0.{cluster_id}.1', 48783))  # random port
+            logger.info("Cluster id acquired: %s", cluster_id)
+            return sock, cluster_id
+        except OSError:
+            continue
+    raise ValueError(f"Couldn't acquire cluster id - looks clusters are not cleared properly")
+
+
+def release_cluster_lock(sock: socket.socket) -> None:
```

Comment:
I don't see any benefit having a wrapper function around the `sock.close()` which could be called directly.
Looks as unneeded obfuscation to me.

Reply:
as above - needed as socket is just implementation. Can change in future and has no importance to cluster.